app/controllers/checkout_controller.rb

```diff
@@ -114,6 +117,17 @@ def redirect_to_cart_path
     end
   end
 
+  def cancel_incomplete_payments
+    # The checkout could not complete due to stock running out. We void any pending (incomplete)
+    # Stripe payments here as the order will need to be changed and resubmitted (or abandoned).
+    @order.payments.pending.each(&:void!)
```

Comment:
Now that #7708 is merged, I think the payment will be in the `requires_authorization` state in OFN. Since this code is being called in the case where the authorization succeeded, we should probably move the payment to the `complete` state (to reflect the successful authorization) and then call `void` on it because the stock is no longer available. We'll want to rebase before testing as well.